system-int-test/test/cmr/system_int_test/ingest/generics_test.clj

```diff
@@ -0,0 +1,199 @@
+(ns cmr.system-int-test.ingest.generics-test
+  "Tests for the Generic Concept Ingest API"
+  (:require
+   [cheshire.core :as json]
+   [clojure.string :as string]
+   [clojure.java.io :as jio]
+   [cmr.common.config :as config]
+   [cmr.common.generics :as gcfg]
+   [cmr.common.util :as cutil]
+   [clojure.test :refer :all]
+   [cmr.ingest.api.generic-documents :as gdocs]
+   [cmr.system-int-test.system :as system]
+   [cmr.system-int-test.utils.ingest-util :as ingest]
+   [cmr.system-int-test.utils.dev-system-util :as dev-sys-util]
+   [cmr.system-int-test.utils.url-helper :as url-helper]
+   [cmr.transmit.config :as transmit-config])
+  (:import
+   [java.util UUID]))
+
+;; TODO: Generic work: known issue, a provider needs to be created through a
+;; fixture, untill this is figured out, manually create a provider before testing
```

Comment:
We need to fix this TODO. This is a pretty serious issue and if it's only known to people who had a direct hand in creating the generics it will cause problems because of lack of domain knowledge when anyone else works with generics.

Reply:
This has been fixed.